drivers/gke/gke_driver.go

```diff
@@ -324,14 +457,81 @@ func getStateFromOpts(driverOptions *types.DriverOptions) (state, error) {
 		d.Locations = append(d.Locations, location)
 	}
 
-	d.NodeCount = options.GetValueFromDriverOptions(driverOptions, types.IntType, "node-count", "nodeCount").(int64)
 	labelValues := options.GetValueFromDriverOptions(driverOptions, types.StringSliceType, "labels").(*types.StringSlice)
 	for _, part := range labelValues.Value {
 		kv := strings.Split(part, "=")
 		if len(kv) == 2 {
-			d.NodeConfig.Labels[kv[0]] = kv[1]
+			d.NodePool.Config.Labels[kv[0]] = kv[1]
+		}
+	}
+	resourceLabelValues := options.GetValueFromDriverOptions(driverOptions, types.StringSliceType, "resource-labels", "resourceLabels").(*types.StringSlice)
+	for _, part := range resourceLabelValues.Value {
+		kv := strings.Split(part, "=")
+		if len(kv) == 2 {
+			d.ResourceLabels[kv[0]] = kv[1]
+		}
+	}
+
+	oauthScopesValues := options.GetValueFromDriverOptions(driverOptions, types.StringSliceType, "oauth-scopes", "oauthScopes").(*types.StringSlice)
+	for _, oauthScope := range oauthScopesValues.Value {
+		d.NodePool.Config.OauthScopes = append(d.NodePool.Config.OauthScopes, oauthScope)
+	}
+
+	// Configuration of Taints
+	taints := options.GetValueFromDriverOptions(driverOptions, types.StringSliceType, "taints").(*types.StringSlice)
+	for _, part := range taints.Value {
+		taint := &raw.NodeTaint{}
+		ekv := strings.Split(part, ":")
+		if len(ekv) == 2 {
+			taint.Effect = ekv[0]
+			kv := strings.Split(ekv[1], "=")
+			if len(kv) == 2 {
+				taint.Key = kv[0]
+				taint.Value = kv[1]
+			}
+		}
+
+		d.NodePool.Config.Taints = append(d.NodePool.Config.Taints, taint)
+	}
+
+	// Configuration of NodePoolAutoscaling
+	d.NodePool.Autoscaling.Enabled = options.GetValueFromDriverOptions(driverOptions, types.BoolType, "enable-nodepool-autoscaling", "enableNodepoolAutoscaling").(bool)
+	d.NodePool.Autoscaling.MinNodeCount = options.GetValueFromDriverOptions(driverOptions, types.IntType, "min-node-count", "minNodeCount").(int64)
+	d.NodePool.Autoscaling.MaxNodeCount = options.GetValueFromDriverOptions(driverOptions, types.IntType, "max-node-count", "maxNodeCount").(int64)
+	d.NodePool.Name = "default-0"
+
+	// Configuration of MasterAuth
+	d.MasterAuth.Username = options.GetValueFromDriverOptions(driverOptions, types.StringType, "master-auth-username", "masterAuthUsername").(string)
```

Comment:
Now that the `master-auth-username` field has been removed from the driver options it should no longer be parsed into the state.  We should just hardcode the state to `admin`.

Reply:
Updated!